### PR TITLE
DB2 JDBC extension

### LINKED ISSUE #6293: Offer DB2 JDBC extension
**Describe the extension**

Offer DB2 JDBC support via extension that works in JVM mode and native mode.

Interested in this extension, please +1 via the emoji/reaction feature of GitHub (top right).

**Configuration suggestion**
```properties
# Add your application.properties here, if applicable.
<typical JDBC configuration properties>
```

**Additional context**
(Add any other context about the proposal here.)

### PR DESCRIPTION
**Pull request by aguibert**
Closes #6293

### CODE REVIEW

File: bom/runtime/pom.xml [hunk 1]
```diff
@@ -538,6 +540,11 @@
                 <artifactId>quarkus-keycloak-admin-client-deployment</artifactId>
                 <version>${project.version}</version>
             </dependency>
+            <dependency>
+                <groupId>io.quarkus</groupId>
+                <artifactId>quarkus-kafka-client</artifactId>
+                <version>${project.version}</version>
```
Comment (gsmet) [inline bom/runtime/pom.xml:546]: This looks totally unrelated. Is it missing from the BOM? If so, should it be added in another PR that we can backport?

File: extensions/jdbc/jdbc-db2/runtime/pom.xml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    <parent>
+        <artifactId>quarkus-jdbc-db2-parent</artifactId>
+        <groupId>io.quarkus</groupId>
+        <version>999-SNAPSHOT</version>
+    </parent>
+    <modelVersion>4.0.0</modelVersion>
+
+    <artifactId>quarkus-jdbc-db2</artifactId>
+    <name>Quarkus - JDBC - DB2 - Runtime</name>
+	<description>DB2 database connector</description>
```
Comment (gsmet) [inline extensions/jdbc/jdbc-db2/runtime/pom.xml:14]: Minor but to be consistent with the others, it should be `Connect to the DB2 database via JDBC`

File: .github/workflows/ci-actions.yml [hunk 3]
```diff
@@ -447,6 +446,7 @@ jobs:
             postgres: "true"
             timeout: 55
             test-modules: >
+              jpa-postgresql
```
Comment (gsmet) [inline .github/workflows/ci-actions.yml:449]: Please add five minutes to the timeout if you're moving things here.

**Review by gsmet — APPROVED** [review]
This looks good to me apart from my minor comment about the timeout. I'll let you fix this.

I let you promote it once we have the GA of the driver.

**Comment by aguibert** [comment]
Looks like CI completed fully green, and I confirmed that the new DB2 tests are now running:
```
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 3.934 s - in io.quarkus.it.jpa.db2.JPAFunctionalityInGraalITCase
[INFO] 
[INFO] Results:
[INFO] 
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0
```

I am going to amend the most recent commit to increase the job timeout from 55m --> 60m as @gsmet requested, but no need to wait for CI all over again just for that trivial change.

Looks ready to merge!
